### PR TITLE
New with file system func

### PR DESCRIPTION
**Pull request by wmarinic**
Added code to create a new aeolic client using a file system

### CODE REVIEW

File: aeolic.go [hunk 1]
```diff
@@ -45,6 +48,19 @@ func NewWithMap(apiKey string, templateMap map[string]string) Client {
 	}
 }
 
+func NewWithFS(apiKey string, dir fs.FS, templateDir string) (Client, error) {
```
Comment (micklove) [inline aeolic.go:43]: NewWithFS is unused?

File: parse.go [hunk 2]
```diff
@@ -33,18 +33,19 @@ func parse(templateName string, templateMap map[string]string, data any) ([]byte
 	return buf.Bytes(), nil
 }
 
-// withTemplates - load templates by file path
-func withTemplates(dirPath string, fileSuffix string) (map[string]string, error) {
+// withTemplates - load templates by directory entry
+func withTemplates(fsys fs.FS, dirPath, fileSuffix string) (map[string]string, error) {
 	rootTemplates := map[string]string{}
-	files, err := os.ReadDir(dirPath)
+
```
Comment (micklove) [inline parse.go:39]: What's the diff here, between `os` and `fs` ? 
(asking coz i don't know :P )

Comment (micklove) [reply parse.go:39]: All good, just read the docs :P

Comment (wmarinic) [reply parse.go:39]: `os` uses the actual directory, whereas with `fs` you can specify which file system you want to use
`os.ReadFile()` was causing issues when using the `embed.FS` and wasn't reading the embedded file

File: aeolic.go (hunk 1 above)
Comment (wmarinic) [reply aeolic.go:43]: The other New funcs have examples in `cmd`, I can add one if needed

**Review by micklove — APPROVED** [review]

File: aeolic.go [hunk 3]
```diff
@@ -20,7 +22,8 @@ type Client struct {
 
 // New - returns a new client with the templates loaded from the provided directory.
 func New(apiKey string, templateDir string) (Client, error) {
-	templates, err := withTemplates(templateDir, ".tmpl.json")
+	fsys := os.DirFS("./")
```
Comment (timwmillard) [inline aeolic.go:25]: Could actually do this now.

```suggestion
	fsys := os.DirFS("./")
	return NewWithFS(apiKey, fsys, templateDir)
```

File: aeolic.go [hunk 4]
```diff
@@ -20,7 +22,8 @@ type Client struct {
 
 // New - returns a new client with the templates loaded from the provided directory.
 func New(apiKey string, templateDir string) (Client, error) {
```
Comment (timwmillard) [inline aeolic.go:24]: Could actually do this now.

```suggestion
func New(apiKey string, templateDir string) (Client, error) {
	fsys := os.DirFS("./")
	return NewWithFS(apiKey, fsys, templateDir)
}
```

**Review by timwmillard — APPROVED** [review]
Nice one

**Pull request merged**